python/mxnet/mxfeatures.py

```diff
@@ -0,0 +1,103 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# coding: utf-8
+# pylint: disable=not-an-iterable
+
+"""runtime detection of compile time features in the native library"""
+
+import ctypes
+import enum
+from .base import _LIB, check_call, mx_uint
+
+feature_names = [
+    "CUDA",
+    "CUDNN",
+    "NCCL",
+    "CUDA_RTC",
+    "TENSORRT",
+    "CPU_SSE",
+    "CPU_SSE2",
+    "CPU_SSE3",
+    "CPU_SSE4_1",
+    "CPU_SSE4_2",
+    "CPU_SSE4A",
+    "CPU_AVX",
+    "CPU_AVX2",
+    "OPENMP",
+    "SSE",
+    "F16C",
+    "JEMALLOC",
+    "BLAS_OPEN",
+    "BLAS_ATLAS",
+    "BLAS_MKL",
+    "BLAS_APPLE",
+    "LAPACK",
+    "MKLDNN",
+    "OPENCV",
+    "CAFFE",
+    "PROFILER",
+    "DIST_KVSTORE",
+    "CXX14",
+    "SIGNAL_HANDLER",
+    "DEBUG"
+]
+
+
+Feature = enum.Enum('Feature', {name: index for index, name in enumerate(feature_names)})
+
+
+def has_feature(feature):
```

Comment:
this seems unnecessary as it would be a simple test of `feature in available_features`, which is just as readable.

Reply:
See comments on linked PR, let's move the discussion there.